### PR TITLE
Schema update

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

The output schema has been updated to be more locked in versus the current broad AnyType. The total output has also been flattened down to PCP's `@interval` list output.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: arcaflow_plugin_pcp/pcp_schema.py [hunk 1]
```diff
@@ -22,13 +22,79 @@ class InputParams:
...
 @dataclass
 class PerfOutput:
     pcp_output: typing.Annotated[
-        typing.Dict[str, typing.Any],
-        schema.name("PCP output dictionary"),
+        typing.List[IntervalOutput],
+        schema.name("PCP output list"),
         schema.description(
-            "Performance data from PCP provided in a " "JSON dictionary format"
+            "Performance data from PCP provided in a " "list format"
```
Comment (jaredoconnell) [inline arcaflow_plugin_pcp/pcp_schema.py]: While you're making this change, you might as well remove the " " that was probably caused by the linter moving two lines into one.

File: arcaflow_plugin_pcp/pcp_plugin.py [hunk 2]
```diff
@@ -171,7 +172,10 @@ def start_pcp(
                     error.cmd[0], error.returncode, error.output
                 )
             )
-        return "success", PerfOutput(pcp_out_json)
+        pcp_metrics_list = pcp_out_json["@pcp"]["@hosts"][0]["@metrics"]
```
Comment (jaredoconnell) [inline arcaflow_plugin_pcp/pcp_plugin.py:175]: Is there ever a case where only getting the first host could be a limitation?

File: arcaflow_plugin_pcp/pcp_schema.py [hunk 3]
```diff
@@ -22,13 +22,79 @@ class InputParams:
...
+        int,
+        schema.id("@interval"),
+        schema.name("Interval ID"),
+        schema.description("The interval ID as reported by PCP"),
+    ]
+    timestamp: typing.Annotated[
+        str,
+        schema.id("@timestamp"),
+        schema.name("Interval timestamp"),
+        schema.description("The timestamp of the reported inverval"),
```
Comment (jaredoconnell) [inline arcaflow_plugin_pcp/pcp_schema.py]: Typo. inverval -> interval

**Review by jaredoconnell — CHANGES_REQUESTED** [review]
See comments.

File: arcaflow_plugin_pcp/pcp_plugin.py (hunk 2 above)
Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_plugin.py:175]: I don't believe so with the current design. Multiple hosts in the output should require PCP itself to be configured to collect data from remote agents. As the plugin stands, we explicitly only collect data from the local node/pod, so there should only ever be a single `host` element in the output.

File: arcaflow_plugin_pcp/pcp_schema.py (hunk 3 above)
Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_schema.py]: fixed

File: arcaflow_plugin_pcp/pcp_schema.py (hunk 1 above)
Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_schema.py]: fixed

**Comment by jaredoconnell** [comment]
The current CI failure is that it just took too long. Once my PR is merged in carpenter, it should not fail unless it takes more than 10 minutes to build.

**Review by jaredoconnell — APPROVED** [review]
It built this time in less than 5 minutes, though the default should now be 10 minutes thanks to the latest PR merged into the image builder.

My local test worked, and looks good.

**Review by portante — APPROVED** [review]

**Pull request merged**